src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -324,67 +317,41 @@ public void setValidateOnlyOverlapping(boolean validateOnlyOverlapping) {
         };
     }
 
-    @Override
-    public int[] getAcceptableTokens() {
-        return getRequiredTokens();
-    }
-
-    @Override
-    public void beginTree(DetailAST rootAST) {
-        frames = new HashMap<>();
-        current.clear();
-
-        final Deque<AbstractFrame> frameStack = new LinkedList<>();
-        DetailAST curNode = rootAST;
-        while (curNode != null) {
-            collectDeclarations(frameStack, curNode);
-            DetailAST toVisit = curNode.getFirstChild();
-            while (curNode != null && toVisit == null) {
-                endCollectingDeclarations(frameStack, curNode);
-                toVisit = curNode.getNextSibling();
-                if (toVisit == null) {
-                    curNode = curNode.getParent();
-                }
-            }
-            curNode = toVisit;
-        }
-    }
-
     @Override
     public void visitToken(DetailAST ast) {
         switch (ast.getType()) {
-            case TokenTypes.IDENT :
+            case TokenTypes.IDENT:
                 processIdent(ast);
                 break;
-            case TokenTypes.CLASS_DEF :
-            case TokenTypes.INTERFACE_DEF :
-            case TokenTypes.ENUM_DEF :
-            case TokenTypes.ANNOTATION_DEF :
-            case TokenTypes.SLIST :
-            case TokenTypes.METHOD_DEF :
-            case TokenTypes.CTOR_DEF :
-            case TokenTypes.LITERAL_FOR :
-                current.push(frames.get(ast));
+            case TokenTypes.CLASS_DEF:
+            case TokenTypes.INTERFACE_DEF:
+            case TokenTypes.ENUM_DEF:
+            case TokenTypes.ANNOTATION_DEF:
+            case TokenTypes.SLIST:
+            case TokenTypes.METHOD_DEF:
+            case TokenTypes.CTOR_DEF:
+            case TokenTypes.LITERAL_FOR:
+                pushFrame(ast);
                 break;
-            default :
+            default:
                 // do nothing
         }
     }
 
     @Override
     public void leaveToken(DetailAST ast) {
         switch (ast.getType()) {
-            case TokenTypes.CLASS_DEF :
-            case TokenTypes.INTERFACE_DEF :
-            case TokenTypes.ENUM_DEF :
-            case TokenTypes.ANNOTATION_DEF :
-            case TokenTypes.SLIST :
-            case TokenTypes.METHOD_DEF :
-            case TokenTypes.CTOR_DEF :
+            case TokenTypes.CLASS_DEF:
```

Comment:
please move formatting to sepparate commit of the same PR, it should not be related to issue and commit should have prefix "minor: formatting".

It is really strange that our chekcstyle rules did not catch this .... please investigate why , we need to enforce no space before ":".

Reply:
Thankfully, @strkkk already did the investigation and he found out that the cause is there is no  `NoWhitespaceBeforeCheck` doesn't support the `COLON` token. 
He has then opened an issue about this matter https://github.com/checkstyle/checkstyle/issues/8045
And I'm already on it. :)

Reply:
ok, thanks a lot, please we wstill need to move formatting changes out of functional changes.

Reply:
I'm doing it right now.

Reply:
Done.